game-app/game-core/src/main/java/games/strategy/engine/framework/startup/launcher/LaunchAction.java

```diff
@@ -6,11 +6,17 @@
 import games.strategy.engine.framework.IGame;
 import games.strategy.engine.framework.LocalPlayers;
 import games.strategy.engine.framework.ServerGame;
+import games.strategy.engine.framework.startup.WatcherThreadMessaging;
+import games.strategy.engine.framework.startup.mc.ServerConnectionProps;
 import games.strategy.engine.framework.startup.mc.ServerModel;
+import games.strategy.engine.framework.startup.ui.PlayerTypes;
 import games.strategy.engine.framework.startup.ui.panels.main.game.selector.GameSelectorModel;
 import games.strategy.engine.player.Player;
+import games.strategy.net.Messengers;
 import java.nio.file.Path;
+import java.util.Optional;
 import java.util.Set;
+import org.triplea.game.chat.ChatModel;
 import org.triplea.sound.ISound;
 
 /**
```

Comment:
As an aside (not for this PR), but I find the name of this class pretty confusing. It seems that the class is mostly responsible for handling game lifecycle events. I'm not sure what the best terminology is for this. In other codebases, I've seen "Client" or "Delegate" used. I know in TripleA, the Delegate terminology is already used and it's not quite the same as this. If we use Client terminology, it could be ServerModelClient.

Or perhaps we can just a different name altogether. For example, ServerBehavior. Thinking about names more, I think ServerModel is also not quite the right name for what that class is doing - it's a bit more of a controller than a model, so perhaps ServerController could be more apt?

... stepping back a bit, the reason I have a problem with LaunchAction is that:
  1. It's not a single action - there's lots of stuff there.
  2. It's not just related to launching.

So even something like ServerActions could be a better name, although not everything is "actions" - e.g. getAutoSaveFile().

Reply:
Agreed. I came up with this name back when I started to migrate a couple of methods that were mainly UI elements related to starting the game.
As more and more things piled up the original name wasn't really fitting anymore but here we are. Really open to anything.